core/domain/question_services.py

```diff
@@ -126,6 +141,32 @@ def get_question_by_id(question_id, strict=True):
         return None
 
 
+def get_question_summaries_linked_to_skills(skill_ids):
+    """Returns the list of question summaries linked to all the skills given by
+    skill_ids.
+
+    Args:
+        skill_ids: list(str). The ids of skills for which the linked questions
+            are to be retrieved.
+
+    Returns:
+        list(QuestionSummary). The list of question summaries linked to the
+            given skill_ids.
+    """
+    skill_ids_check_dict = {}
+    for skill_id in skill_ids:
+        skill_ids_check_dict[skill_id] = True
+
+    question_ids = []
+    all_models = question_models.QuestionSkillLinkModel.get_all()
```

Comment:
Um, no, this is too much. Pretend you're dealing with 100k questions :)

Do a query/filter (talk to @nithusha21 if you need to, he knows how this works). Encapsulate that within the storage model in a get_model_instances_with_skill_id(skill_id).

Reply:
I had thought of that, but here I need to fetch all QuestionSkillLink models whose skill_id field has a value present in the input array and couldn't find an 'in array' check for datastore models, so did this.
Or should I iterate through input array, and do a query-fetch individually for each skill_id?

Reply:
Try checking out the IN filter command. Maybe that can help? https://cloud.google.com/appengine/docs/standard/python/ndb/queries#neq_and_in

Reply:
Thanks! @nithusha21 . Yeah, the 'IN' operation worked.

Reply:
Done.